airbyte-cdk/python/airbyte_cdk/sources/declarative/extractors/record_selector.py

```diff
@@ -38,8 +42,29 @@ def select_records(
         next_page_token: Optional[Mapping[str, Any]] = None,
     ) -> List[Record]:
         all_records = [Record(data, stream_slice) for data in self.extractor.extract_records(response)]
+        filtered_records = self._filter(all_records, stream_state, stream_slice, next_page_token)
+        self._transform(filtered_records, stream_state, stream_slice)
```

Comment:
Don't we have all the information the transform have in the filter (and even more since we have `next_page_token` also)? In that case, I think we can do anything in the filter we can do in the transform

Reply:
but the transformations are done after the filter, right? This means we can't filter on fields added by a transformation

Reply:
I'll merge since this is out-of-scope for the current work but we can still continue the discussion

Reply:
that makes sense. I don't expect any existing connector to break since the transformations were already applied afterwards